src/main/java/controllers/ProjectInputController.java

```diff
@@ -333,18 +333,25 @@ public ProjectInputController(ProjectRepository projectRepository) {
     public String[] projectEditTaskRequirements(Project projectToManage, String projectCommand) {
         ArchDukeLogger.logDebug(ProjectInputController.class.getName(),
                 "[projectEditTaskRequirements] User input: '" + projectCommand + "'");
+        if (projectCommand.length() <= 23) {
+            return new String[] {"Task index is missing! Please input index of task to be edited!"};
+        }
         try {
-            int taskIndexNumber = Integer.parseInt(projectCommand.substring(23).split(" ")[0]);
-            String updatedTaskRequirements = projectCommand.substring(projectCommand.indexOf("-"));
+            int taskIndexNumber = Integer.parseInt(projectCommand.substring(23).trim().split(" ")[0]);
             if (projectToManage.getNumOfTasks() >= taskIndexNumber && taskIndexNumber > 0) {
-                projectToManage.editTaskRequirements(taskIndexNumber,updatedTaskRequirements);
-                return new String[] {"The requirements of your specified task has been updated!"};
+                if (!projectCommand.contains("-")) {
+                    return new String[] {"No flags are found! Please use flags such as '-r' or '-rm' to indicate "
+                            + "the new requirements to be added or removed! Refer to the user guide for more help!"};
+                } else {
+                    String updatedTaskRequirements = projectCommand.substring(projectCommand.indexOf("-"));
```

Comment:
Might have an error here if the line does not contain "-"!

Reply:
if the line doesn't have "-",  it will enter the if block and return the message that no flags are found